src/components/multi_select_state_utils.js

```diff
@@ -32,11 +32,55 @@ export const getSelectedByAllItems = (itemsToSelect, selectedItems, items) => {
   return [...destinationItems, ...sourceItems];
 };
 
-export const getMinMaxIndexes = (currentIndex, firstItemShiftSelected) =>
+export const getMinMaxIndexes = (
+  currentIndex,
+  firstItemShiftSelected,
+  items,
+  selectedItems,
+  maxSelected
+) => {
+  const outsideSelected = getSelectedItemsOutsideInterval(
+    currentIndex,
+    firstItemShiftSelected,
+    items,
+    selectedItems
+  ).length;
+  const shouldBeSelect =
+    outsideSelected + Math.abs(firstItemShiftSelected - currentIndex) + 1;
+
+  if (maxSelected && maxSelected <= shouldBeSelect) {
```

Comment:
can u extract this logic to function with clear name, something like: getIndexesWhenShiftSelected ?

Reply:
Updated